### ISSUE TITLE
SVG file converted at 90dpi, not 72dpi, resulting in bigger PDF files

### ISSUE THREAD

**Issue by elQuero93** [jgm/pandoc#5721]
I've found that the conversion of SVG images outputs a PDF of wrong size due to the fact that SVG file is converted at 90dpi resolution.
Pandoc uses the following command for conversion, specified in `src/Text/Pandoc/PDF.hs`:

```
rsvg-convert -f pdf -a -o <output_file> <input_file>
```

while the following one will output a correctly sized PDF file:

```
rsvg-convert -f pdf -a -o --dpi-x 72 --dpi-y 72 <output_file> <input_file>
```

I'm on "pandoc 2.7.3 - Compiled with pandoc-types 1.17.5.4, texmath 0.11.2.2, skylighting 0.8.2".

**Comment by jgm** [comment]
I gather then that rsvg-convert uses a default of 90 dpi?  Why is 72dpi the right choice here?

**Comment by mb21** [comment]
Since svgs are vector graphics, dpi is usually not important. Rather how big the image is in the pdf is determined by how it's placed in the latex document...

Can you post a minimal example input file (markdown or latex), along with an example svg?

**Comment by agusmba** [comment]
> Since svgs are vector graphics, dpi is usually not important. Rather how big the image is in the pdf is determined by how it's placed in the latex document...
> 
I'm guessing the dpi is used for the raster output image from `rsvg-convert` and could have an indirect impact on how big the image is in the resulting PDF.

**Comment by jgm** [comment]
@elQuero93 can you respond to the query above?  Why is 72dpi the right choice here?

**Comment by alerque** [comment]
I think this is almost certainly we should have a command line flag for to pass through to conversion filters like `rsvg`, but hard coding a lower resolution is probably not the answer. In fact by default I think leaving this to `rsvg`'s out of the box settings is probably the best thing to do. Converting vector graphics to raster is inherently a lossy operation, and how lossy it should be by default for a given output format is highly controversial.

72 DPI is an old "standard" for raster graphics on the internet. Most browsers before this decade used this as their default between pixel and other units of measurement by default. There has been some movement to change this, usually via progressive enhancement, but not a blanket rule.

SVG is a bit different. In the early days apps adopted a mix of default "resolutions" that set their output scaling. Some did use 72. More used 90. Lately many have been switching from 90 to 96 because of CSS where the latter resolution has become the defacto norm. (Note even browsers that scale raster graphics by default at 72 DPI will make CSS based calculations at 96 DPI.

TL;DR: This whole subject is a mess and I really don't think Pandoc should be "opinionated" by hard coding a value. If there isn't a way to pass this as an option to `rsvg-convert` there should be, but `rsvg`'s default of 90 DPI is there for a reason and if anything the argument could be made that it should be 96, but not 72.

**Comment by mb21** [comment]
I was about to close this.... but would it make sense that pandoc passes on the [`--dpi` option](https://pandoc.org/MANUAL.html#option--dpi) to rsvg somehow?

Currently, I think `--dpi` it's only used for images, as seen in https://pandoc.org/MANUAL.html#extension-link_attributes

**Comment by alerque** [comment]
@mb21 Yes I think that would make sense. Even if formats that can embed the SVG data so there is no rasterization involved it will still affect size calculations and hence the layout.

**Comment by jgm** [comment]
Agreed, that's a good idea.  `writerDpi` in `WriterOptions` should be easily available in that context.

**Comment by jgm** [comment]
Note that the default value is 96, so this would mean a change to default behavior.

**Comment by elQuero93** [comment]
Hi, I'm sorry for having ignored this thread but I didn't received notifications and I've forgot about it.

I'm here to report that this functionality still does not work correctly IMHO, but I think that this time the issue is up to rsvg-converter.

SVG drawing size can be specified in real-world units, like millimeters, or computer-screen units, like pixels. Here you can find two SVG drawings, one 50x50mm, the other

**Comment by elQuero93** [comment]
Hi, I'm sorry for having ignored this thread but I didn't received notifications and I've forgot about it.

I've tested the functionality introduced by the [a64b3ab](https://github.com/jgm/pandoc/commit/a64b3ab61ff34dca25f04f4f97b283dc5159ac87) commit and I think that it still does not work correctly, but this time is due to an issue in rsvg-convert.

SVG drawing size can be specified in real-world units, like inches, or computer-screen units, like pixels. I've tried to convert two SVGs, one 4x4in (`rect_in.svg`) and another 384x384px (`rect_px.svg`), both at 72dpi and 96dpi.
What I expect to obtain is that for SVG specified in real-world unit the resulting PDF will be of the same size (so `--dpi-x` and `--dpi-y` parameters will be ignored), while for SVG specified in computer-screen units resulting PDF size must change.
Instead of what I've described above I obtain all wrongly sized PDFs. Here is a detailed list of what I obtain:

- `rect_in.svg` @72dpi: ~5.3x5.3in PDF -> should be 4x4in;
- `rect_in.svg` @96dpi: ~5.3x5.3in PDF -> should be 4x4in;
- `rect_px.svg` @72dpi: 4x4in PDF -> should be ~5,3x5,3in;
- `rect_px.svg` @96dpi: ~5.3x5.3in PDF -> should be 4x4in;

Here you can find input and output files: [test_files.zip](https://github.com/jgm/pandoc/files/3683415/test_files.zip). I'm on rsvg-convert 2.44.15.

Do you agree? If yes I'll submit an issue against rsvg-convert.

The only good news is that the pandoc `--dpi` parameter will be ignored for real-world unit sized SVG, as with raster images explicitly defining DPI in their metadata (P.S.: this behavior is not documented in the pandoc manual. Maybe it should?)

**Comment by jgm** [comment]
> Do you agree? If yes I'll submit an issue against rsvg-convert.

You're probably a better judge than me of what the behavior should be. Why don't you see if the rsvg-convert devs agree that it's a bug.

> P.S.: it seems that the pandoc --dpi parameter is ignored if raster images specifies a DPI value in their metadata, but this is not reported in pandoc manual. Is it an expected behavior?

What format? Can you give an example?

**Comment by elQuero93** [comment]
> What format? Can you give an example?

Here you are: [DPI raster images tests.zip](https://github.com/jgm/pandoc/files/3694591/DPI.raster.images.tests.zip).

In this archive you can find some tests against PNG and JPG images, both at 96dpi and 72dpi and, as you can see in output PDF files, `--dpi` value is ignored.
I didn't found any way to remove images DPI info, so I can't test pandoc behaviour in that case.

Tests have been done on pandoc 2.8 (compliled from master few days on October 2nd) and 2.7.3 (release version).

**Comment by jgm** [comment]
Ah, I see.  Well, here pandoc is simply passing the jpg through to a latex `\includegraphics` command; no size conversions are done.  The docs say that the `--dpi` option is used for converting between pixels and other measures, such as inches.  That isn't called for in your example.

In cases where measures are provided, pandoc will always take the dpi from the image itself, if it has this information.  The `--dpi` command-line option is provided mainly for cases where there the image  doesn't contain this information.
